cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -72,7 +72,8 @@ impl ArgAssigner for Args {
         // Check for a legal type.
         // We don't support SIMD yet, so break all vectors down.
         if ty.is_vector() {
-            return ValueConversion::VectorSplit.into();
+            return ArgAction::Assign(ArgumentLoc::Unassigned);
+            //return ValueConversion::VectorSplit.into();
```

Comment:
@bnjbvr, I think something along these lines is going to be necessary to get rid of all of the `vsplit`-ing prior to the EBB return. I realized that x86/abi.rs:173 was calling `legalize_args` which calls into `ArgAssigner::assign` (this snippet). Instead of asking cranelift to do a `ValueConversion` of some form I would expect that we want to keep the vector value in a register. Perhaps something like the code down at 128-132 may be on the right track but I wasn't really sure. This change causes an error (`FAIL filetests/isa/x86/legalize-simd.clif: panicked in worker #3: function return signature should be legalized`) but I thought I would leave it in so you could see where I stopped.

Reply:
Agreed something like the code around 128-132 might work; maybe this block could be removed and the block handling float types below could handle this too.